shell/platform/android/io/flutter/embedding/engine/FlutterJNI.java

```diff
@@ -120,9 +120,14 @@ public static native void nativeInit(
    */
   public static native void nativePrefetchDefaultFontManager();
 
-  // TODO(mattcarroll): add javadocs
+  private native boolean nativeGetIsSoftwareRenderingEnabled();
+
+  @VisibleForTesting
```

Comment:
Can this be package private?

Reply:
It looks like it's not just visible for testing - it's used below in `FlutterRenderer.isSoftwareRenderingEnabled()`

Reply:
right on. Fixed